pkg/dao/types/workflow.go

```diff
@@ -83,26 +86,46 @@ func NewWorkflowStepApprovalSpec(spec map[string]any) (*WorkflowStepApprovalSpec
 		return nil, errors.New("invalid input: invalid approval type")
 	}
 
-	if v, ok := spec[WorkflowStepApprovalUsers].([]any); ok {
-		users, err := toObjectIDs(v)
-		if err != nil {
-			return nil, err
-		}
-		s.ApprovalUsers = removeDuplicatedUsers(users)
+	userIndexes := []string{
+		WorkflowStepApprovalUsers,
+		WorkflowStepApprovedUsers,
+		WorkflowStepRejectedUsers,
 	}
 
-	if v, ok := spec[WorkflowStepApprovedUsers].([]any); ok {
-		users, err := toObjectIDs(v)
-		if err != nil {
-			return nil, err
+	for i := range userIndexes {
+		if _, ok := spec[userIndexes[i]]; !ok {
+			continue
+		}
+
+		if v, ok := spec[userIndexes[i]].([]any); ok {
+			users, err := toObjectIDs(v)
+			if err != nil {
+				return nil, err
+			}
+
+			switch userIndexes[i] {
+			case WorkflowStepApprovalUsers:
+				s.ApprovalUsers = removeDuplicatedUsers(users)
+			case WorkflowStepApprovedUsers:
+				s.ApprovedUsers = removeDuplicatedUsers(users)
+			case WorkflowStepRejectedUsers:
+				s.RejectedUsers = removeDuplicatedUsers(users)
+			}
 		}
-		s.ApprovedUsers = removeDuplicatedUsers(users)
 	}
 
 	return s, nil
 }
 
+func (s *WorkflowStepApprovalSpec) IsRejected() bool {
+	return len(s.RejectedUsers) > 0
+}
+
 func (s *WorkflowStepApprovalSpec) IsApproved() bool {
+	if s.IsRejected() {
+		return false
+	}
```

Comment:
When user config or here, then one user approved and one user rejected, it should be approved right? But based on the code it would return not approved

Reply:
If one user reject the step, the step is rejected, no matter how many other users approved.